src/Resources/Resources.Test/ScenarioTests/TagTests.ps1

```diff
@@ -0,0 +1,266 @@
+# ----------------------------------------------------------------------------------
+
+<#
+.SYNOPSIS
+Utility method to test CreeteOrUpdate for Tags within tracked resources and subscription.
+#>
+function Test-TagCreateOrUpdateWithResourceIdParams($resourceId)
+{
+    # Setup
+    $expected = @{"key1"="value1"; "key2"="value2";}
+
+    try
+    {
+        # Test
+        $res = New-AzTag -ResourceId $resourceId -Tag $expected
+        [hashtable]$actual = $res.Properties.TagsProperty
+
+        # Assert
+        Assert-AreEqual $expected.Count $actual.Count
+        Assert-AreEqual $expected.key1 $actual.key1
+        Assert-AreEqual $expected.key2 $actual.key2
+    }
+    finally
+    {
+        # Cleanup
+        # Clean-TagsByResourceId $resourceId
+        Remove-AzTag -ResourceId $resourceId
+    } 
+}
+
+<#
+.SYNOPSIS
+Tests creating or updating tags on subscription.
+#>
+function Test-TagCreateOrUpdateWithResourceIdParamsForSubscription
+{
+    # Setup
+    # $subs = Get-AzSubscription
+    # $subId = $subs[0].Id
+    $subscriptionId = "b9f138a1-1d64-4108-8413-9ea3be1c1b2d"
+    $resourceId = "/subscriptions/" + $subscriptionId
+
+    Test-TagCreateOrUpdateWithResourceIdParams $resourceId
+   
+}
+
+<#
+.SYNOPSIS
+Tests creating or updating tags on tracked resource.
+#>
+function Test-TagCreateOrUpdateWithResourceIdParamsForResource
+{
+    # Setup
+    $resourceId = "/subscriptions/b9f138a1-1d64-4108-8413-9ea3be1c1b2d/resourcegroups/TagsApiSDK/providers/Microsoft.Compute/virtualMachines/TagTestVM";
+
+    Test-TagCreateOrUpdateWithResourceIdParams $resourceId
+}
+
+
+<#
+.SYNOPSIS
+Utility method to test updating tags on subscription and tracked resource, including Merge, Replace, and Delete Operation.
+#>
+function Test-UpdateWithResourceIdParams($resourceId)
+{
+    # Setup
+    $original = @{"key1"="value1"; "key2"="value2";}
+    New-AzTag -ResourceId $resourceId -Tag $original
+
+    try
+    {
+        # Test
+        {
+            # merge operation
+            $merged = @{"key1"="value1"; "key3"="value3";}
+            $res = Update-AzTag -ResourceId $resourceId -Tag $merged -Operation Merge
+
+            $expected = @{"key1"="value1"; "key2"="value2"; "key3"="value3";}
+            [hashtable]$actual = $res.Properties.TagsProperty
+
+            # Assert
+            Assert-AreEqual $expected.Count $actual.Count
+            Assert-AreEqual $expected.key1 $actual.key1
+            Assert-AreEqual $expected.key2 $actual.key2
+            Assert-AreEqual $expected.key3 $actual.key3
+        }
+
+        {
+            # repalce operation
+            $replaced = @{"key1"="value1"; "key3"="value3";}
+            $res = Update-AzTag -ResourceId $resourceId -Tag $replaced -Operation Replace
+
+            $expected = $replaced
+            [hashtable]$actual = $res.Properties.TagsProperty
+
+            # Assert
+            Assert-AreEqual $expected.Count $actual.Count
+            Assert-AreEqual $expected.key1 $actual.key1
```

Comment:
compare value as well .. value should be check with case sensitivity, key is case insensitive